### PR TITLE
feat(chart): add priorityClassName override template and value

### PR DESCRIPTION
**Pull request by niladrih**

### CODE REVIEW

**Comment by niladrih** [comment]
bors try

File: k8s/upgrade-job/src/common/error.rs [hunk 1]
```diff
@@ -438,6 +438,10 @@ pub(crate) enum Error {
         context
     ))]
     NoNamespaceInPod { pod_name: String, context: String },
+
+    /// Error for when the thin-provisioning option are absent, but still tried to fetch it.
```
[the remark this answers is not in the document]
Comment (niladrih) [reply k8s/upgrade-job/src/common/error.rs:442]: Damn.

**Review by avishnu — APPROVED** [review]

File: k8s/upgrade-job/src/helm/chart.rs [hunk 2]
```diff
@@ -100,6 +111,18 @@ impl CoreValues {
     pub(crate) fn core_thin_volume_commitment_initial(&self) -> Result<String> {
         self.agents.core_thin_volume_commitment_initial()
     }
+
+    pub(crate) fn priority_class_name_is_absent(&self) -> bool {
```
Comment (tiagolobocastro) [inline k8s/upgrade-job/src/helm/chart.rs:115]: missing doc comments

**Review by tiagolobocastro — APPROVED** [review]

**Comment by niladrih** [comment]
bors merge

**Pull request merged**